dlink-admin/src/main/java/com/dlink/service/impl/StudioServiceImpl.java

```diff
@@ -101,22 +97,29 @@ public class StudioServiceImpl implements StudioService {
 
     private static final Logger logger = LoggerFactory.getLogger(StudioServiceImpl.class);
 
-    @Autowired
-    private ClusterService clusterService;
-    @Autowired
-    private ClusterConfigurationService clusterConfigurationService;
-    @Autowired
-    private SavepointsService savepointsService;
-    @Autowired
-    private DataBaseService dataBaseService;
-    @Autowired
-    private TaskService taskService;
-
-    @Autowired
-    private FragmentVariableService fragmentVariableService;
-
-    @Autowired
-    private UDFService udfService;
+    private final ClusterService clusterService;
+    private final ClusterConfigurationService clusterConfigurationService;
+    private final SavepointsService savepointsService;
+    private final DataBaseService dataBaseService;
+    private final TaskService taskService;
+    private final FragmentVariableService fragmentVariableService;
+    private final UDFService udfService;
+
+    public StudioServiceImpl(ClusterService clusterService,
+                             ClusterConfigurationService clusterConfigurationService,
+                             SavepointsService savepointsService,
+                             DataBaseService dataBaseService,
+                             TaskService taskService,
+                             FragmentVariableService fragmentVariableService,
+                             UDFService udfService) {
+        this.clusterService = clusterService;
+        this.clusterConfigurationService = clusterConfigurationService;
+        this.savepointsService = savepointsService;
+        this.dataBaseService = dataBaseService;
+        this.taskService = taskService;
+        this.fragmentVariableService = fragmentVariableService;
+        this.udfService = udfService;
+    }
```

Comment:
Why construct it?

Reply:
> Why construct it?

@Autowired is not recommended by the office, [Reason](https://betulsahinn.medium.com/why-is-autowired-annotation-not-recommended-4939c46da1f8)